text/0000-template.md

```diff
@@ -0,0 +1,341 @@
+
+# Summary
+[summary]: #summary
+
+Version 2 of the protocol to synchronize the fullnode with the p2p network.
+
+# Motivation
+[motivation]: #motivation
+
+Sync-v1 propagates arbitrary orphan blocks that do not add to the network in any way and waste resources. More critically it will _have_ to repropagate all those orphan blocks for the network to converge.
+
+Sync-v2 uses many lessons from sync-v1. It is faster to sync, requires less indexing and uses less CPU.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The goal of the sync-v2 is to get the fullnode synchronized (synced) with the p2p network. As the network is decentralized and all nodes are equal, it happens that the fullnode must individually sync with each peer it is connected to. Even though we expect to have all peers agreeing in which is the bestchain, there might be one or more peers with different bestchains.
+
+A checkpoint is a block that must be part of the bestchain, i.e. it cannot be orphan. It means that the bestchain must contain all checkpoints. The first natural and obvious checkpoint is the genesis with height zero. Sync-v2 supports static checkpoints, i.e., the list of checkpoints is hard-coded in the source code and cannot modified. There are several benefits to have support for checkpoints, including to prevent a large reorg attacks when the hashrate is not that big.
+
+Notice that we wouldn't need to have a list of checkpoints because the last checkpoint itself uniquely defines the chain until the genesis block. But there are performance gains in the sync algorithm knowing intermediate checkpoints beforehand.
+
+We say that a transaction belongs to the first block of the bestchain that verifies it. Transactions without a first block are said to be part of the mempool. Notice that, in case of a reorg, a block may turn into an orphan block and its transactions will belong to the mempool.
+
+From now on, let's enumerate the blocks of the bestchain by their heights (e.g., `b0` is the genesis, `b1` is the block with height one, `b15` is the block with height 15, and so on and so forth). We can also use the following notation: `[b0, b1, ..., b15, ...]` to indicate a sequence of blocks connected by their parents.
+
+Sync-v2 uses the fact that the weight of the blocks is way bigger than the transactions'. The synchronization happens in four phases: (i) sync the blocks until last checkpoint, (ii) sync the blocks from the last checkpoint to the last best block, (iii) sync the mempool, and finally (iv) real-time propagation.
+
+## Phase 1: From genesis to the last checkpoint
+
+The first phase syncs all blocks (and their transactions) from genesis until last checkpoint. In this phase, we can safely download blocks from a given checkpoint and moving backwards (i.e., decreasing the height). For example, if we know that `b100` is a checkpoint, we can safely download `b99`, and then `b98`, and then `b97`, and so on.
```

Comment:
> and *start* moving

or

> and move

Anyway, just "moving" reads a bit odd. Not sure if it's wrong though.

Reply:
Thanks! I agree it is a bit odd, and I changed to "start going".